libtesting-ui/src/main/java/com/mapbox/navigation/testing/ui/DualLocationMocker.kt

```diff
@@ -0,0 +1,38 @@
+package com.mapbox.navigation.testing.ui
+
+import android.location.Location
+
+/**
+ * Mocker that pushes updates to both location provider implementations
+ * so that the sample is available regardless where the running location engine tries to get it from.
+ *
+ * Introduction of this dual mocker was needed to initially work around CORESDK-1528
+ * so that the last location is available in both Google fused and Android location providers at the same time.
+ */
+internal class DualLocationMocker(
+    private val systemLocationMocker: SystemLocationMocker,
+    private val fusedLocationMocker: FusedLocationMocker?,
+) : LocationMocker {
+    override fun before() {
+        systemLocationMocker.before()
+        fusedLocationMocker?.before()
+    }
+
+    override fun after() {
+        systemLocationMocker.after()
+        fusedLocationMocker?.after()
+    }
+
+    override fun mockLocation(location: Location) {
+        systemLocationMocker.mockLocation(location)
+        fusedLocationMocker?.mockLocation(
+            location.apply {
+                provider = FusedLocationMocker.DEFAULT_PROVIDER_NAME
+            }
+        )
+    }
+
+    override fun generateDefaultLocation(): Location {
+        return systemLocationMocker.generateDefaultLocation()
```

Comment:
Doesn't this break something? We used to have default location with "fused" provider, now it will be "gps".

Reply:
In practice it shouldn't because we push locations to both providers at the same time. Ideally, of course, if there was no bug in the Common SDK we wouldn't have to do that in the first place.

Reply:
We do push it to both providers, but now the fused provider will return the location with provider="gps", won't it?